spec/lib/contextual_logger_spec.rb

```diff
@@ -115,6 +116,38 @@ def expect_log_line_to_be_written(log_line)
         expect(console_log_stream.string.gsub(/\[[^]]+\]/, '[]')).to eq("E, [] ERROR -- {:service=>\"test_service\"}: error message\n")
       end
     end
+
+    describe 'with nil message' do
+      it "logs null" do
+        logger.level = Logger::Severity::DEBUG
+        log_message_at_every_level(logger, nil)
+
+        expect(log_stream.string.gsub(/"timestamp":"[^"]+"/, '<time>')).to eq(<<~EOS)
+          {"message":null,"severity":"DEBUG",<time>}
+          {"message":null,"severity":"INFO",<time>}
+          {"message":null,"severity":"WARN",<time>}
+          {"message":null,"severity":"ERROR",<time>}
+          {"message":null,"severity":"FATAL",<time>}
+          {"message":null,"severity":"ANY",<time>}
```

Comment:
@jebentier Was it intentional that the non-string message gets passed right into JSON like this? I was expecting it get formatted into a string, as ::Logger's DefaultFormatter does:
https://github.com/ruby/logger/blob/master/lib/logger/formatter.rb#L32
(Too bad that `msg2str` method is an instance method rather than an eigenclass method. That means we can't just call it.)

Reply:
Yes. The default formatting for the contextual logger is structured JSON. Outputing in other formats is configurable, but the purpose of the gem is to produce structured, tagged logging.

Reply:
Ahh read your second comment and I see what you’re asking now. Yes let’s make the improvement that message is always a string to keep things standardized and structured in a specific redictable manner.

Reply:
Ok, I guess I'll make a copy of `msg2str` then. Or, there is an alternative to share that code: I could create a constant that is a `Logger::Formatter`. But then it's a private method, so I'd have to `send` to it. Or I suppose I could inherit from it and expose expose an indirect . Probably not worth jumping through all those hoops though, vs just copying this:
```
    def msg2str(msg)
      case msg
      when ::String
        msg
      when ::Exception
        "#{ msg.message } (#{ msg.class })\n#{ msg.backtrace.join("\n") if msg.backtrace }"
      else
        msg.inspect
      end
    end
```

Reply:
Agreed, I think copying the code is a better approach here.  Maybe also give it a more human readable name too. No need to make it _exactly_ the same as `msg2str` since it's not the same 😉

Reply:
I'll definitely leave a comment pointing back to the original source. And of course we'll have have tests. `nil` and `false` are already tested (but will change). I'll also add a symbol and an object with a custom `inspect` and also an `Exception`.